packages/icon-button/src/mwc-icon-button-base.ts

```diff
@@ -25,15 +28,90 @@ export class IconButtonBase extends LitElement {
 
   @property({type: String}) label = '';
 
+  @query('button') buttonElement!: HTMLElement;
+
+  @queryAsync('mwc-ripple') ripple!: Promise<Ripple|null>;
+
+  @internalProperty() protected shouldRenderRipple = false;
+
+  protected rippleHandlers: RippleHandlers = new RippleHandlers(() => {
+    this.shouldRenderRipple = true;
+    return this.ripple;
+  });
+
+  protected renderRipple() {
+    if (this.shouldRenderRipple) {
+      return html`<mwc-ripple .disabled="${
+          this.disabled}" unbounded></mwc-ripple>`;
+    }
+    return html``;
+  }
+
+  focus() {
+    const buttonElement = this.buttonElement;
+    if (buttonElement) {
+      this.rippleHandlers.startFocus();
+      buttonElement.focus();
+    }
+  }
+
+  blur() {
+    const buttonElement = this.buttonElement;
+    if (buttonElement) {
+      this.rippleHandlers.endFocus();
+      buttonElement.blur();
+    }
+  }
+
   /** @soyCompatible */
   protected render() {
     return html`<button
-    .ripple="${ripple()}"
-    class="mdc-icon-button"
-    aria-label="${this.label || this.icon}"
-    ?disabled="${this.disabled}">
+        class="mdc-icon-button"
+        aria-label="${this.label || this.icon}"
+        ?disabled="${this.disabled}"
+        @focus="${this.handleRippleFocus}"
+        @blur="${this.handleRippleBlur}"
+        @mousedown="${this.handleRippleActivate}"
+        @mouseenter="${this.handleRippleMouseEnter}"
+        @mouseleave="${this.handleRippleMouseLeave}"
+        @touchstart="${this.handleRippleActivate}"
+        @touchend="${this.handleRippleDeactivate}"
+        @touchcancel="${this.handleRippleDeactivate}">
+      ${this.renderRipple()}
     <i class="material-icons">${this.icon}</i>
     <slot></slot>
   </button>`;
   }
+
+  @eventOptions({passive: true})
+  private handleRippleActivate(evt?: Event) {
```

Comment:
also can these be made protected

Reply:
Also style guide asks to use full word `event` which gets an eyeroll from me every time